### PR TITLE
Move API doc descriptions inside the records

### PR DESCRIPTION
**Pull request by TharmiganK**
## Purpose

> Related to [`HATEOAS link support #1637`](https://github.com/ballerina-platform/ballerina-standard-library/issues/1637#issuecomment-895066495)

## Checklist
- [x] Linked to an issue
- [ ] Updated the changelog
- [ ] Added tests

### CODE REVIEW

**Review by chamil321 — APPROVED** [review]

File: ballerina/http_types.bal [hunk 1]
```diff
@@ -103,21 +103,19 @@ public type CommonClientConfiguration record {|
...
 # Represents a server-provided hyperlink
-#
-# + rel - Link relation type which describes how the current context (source) is related to the target resource
-# + href - Target URI
-# + types - Expected resource media types
-# + methods - Expected resource methods
 public type Link record {
+    # Link relation type which describes how the current context (source) is related to the target resource
     string rel;
+    # Target URI
```
Comment (shafreenAnfar) [inline ballerina/http_types.bal:109]: Target URL

File: ballerina/http_types.bal [hunk 2]
```diff
@@ -103,21 +103,19 @@ public type CommonClientConfiguration record {|
...
     string href;
+    # Expected resource media types
     string[] types?;
+    # Expected resource methods
```
Comment (shafreenAnfar) [inline ballerina/http_types.bal:113]: Allowed resource methods

File: ballerina/http_types.bal [hunk 3]
```diff
@@ -103,21 +103,19 @@ public type CommonClientConfiguration record {|
...
-# + rel - Link relation type which describes how the current context (source) is related to the target resource
-# + href - Target URI
-# + types - Expected resource media types
-# + methods - Expected resource methods
 public type Link record {
+    # Link relation type which describes how the current context (source) is related to the target resource
     string rel;
+    # Target URI
     string href;
+    # Expected resource media types
```
Comment (shafreenAnfar) [inline ballerina/http_types.bal:111]: Expected resource representation media types

File: ballerina/http_types.bal [hunk 4]
```diff
@@ -103,21 +103,19 @@ public type CommonClientConfiguration record {|
...
 
 # Represents a server-provided hyperlink
-#
-# + rel - Link relation type which describes how the current context (source) is related to the target resource
-# + href - Target URI
-# + types - Expected resource media types
-# + methods - Expected resource methods
 public type Link record {
+    # Link relation type which describes how the current context (source) is related to the target resource
     string rel;
```
Comment (shafreenAnfar) [inline ballerina/http_types.bal:108]: Names a relationship of the linked target to the current representation

**Review by shafreenAnfar — APPROVED** [review]

**Pull request merged**